public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/DBClusterBasicOptions/DBClusterBasicOptions.tsx

```diff
@@ -18,35 +18,51 @@ import { isOptionEmpty } from '../../DBCluster.utils';
 import { useDatabaseVersions } from './DBClusterBasicOptions.hooks';
 import { CLUSTER_NAME_MAX_LENGTH } from './DBClusterBasicOptions.constants';
 import { getDatabaseOptionFromOperator } from '../../../Kubernetes/Kubernetes.utils';
-import { Operator } from '../../../Kubernetes/Kubernetes.types';
+import { Kubernetes, Operator } from '../../../Kubernetes/Kubernetes.types';
+
+const getAvailableDatabaseOptions = (kubernetesCluster: Kubernetes): DatabaseOption[] => {
+  const { operators } = kubernetesCluster;
+  const availableDatabaseOptions: DatabaseOption[] = [];
+
+  Object.entries(operators).forEach(([operator, { status }]: [string, Operator]) => {
+    if (status === KubernetesOperatorStatus.ok) {
+      availableDatabaseOptions.push(getDatabaseOptionFromOperator(operator as Operators) as DatabaseOption);
+    }
+  });
+
+  return availableDatabaseOptions;
+};
 
 export const DBClusterBasicOptions: FC<DBClusterBasicOptionsProps> = ({ kubernetes, form }) => {
   const { required, maxLength } = validators;
   const { change } = form;
   const { kubernetesCluster, databaseType } = form.getState().values;
   const [databaseVersions, setDatabaseVersions] = useState<SelectableValue[]>([]);
   const [loadingDatabaseVersions, setLoadingDatabaseVersions] = useState(false);
+  const firstRender = useRef(true);
   const onChangeDatabase = useCallback((databaseType) => {
     if (databaseType.value !== Databases.mysql) {
       change(AddDBClusterFields.topology, DBClusterTopology.cluster);
     }
 
     change(AddDBClusterFields.databaseType, databaseType);
+    form.mutators.setClusterName(databaseType.value);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   const kubernetesOptions = getKubernetesOptions(kubernetes);
 
-  const [databaseOptions, setDatabaseOptions] = useState(DATABASE_OPTIONS);
-  const onChangeCluster = useCallback((selectedKubernetes) => {
-    const { operators } = selectedKubernetes;
-    const availableDatabaseOptions: DatabaseOption[] = [];
+  const [databaseOptions, setDatabaseOptions] = useState(() => {
+    if (firstRender.current && kubernetesCluster) {
```

Comment:
Is the `firstRender` ref needed? I think the initial value function will only get called on the first render, so the ref may be redundant here.

Reply:
yes, @matejkubinec, you are totally right, I've fixed